src/filesystem.ts

```diff
@@ -105,6 +115,65 @@ export async function loadFileSystem({ config, dependencies, rootKey, username }
   return fs
 }
 
+/**
+ * Recover a user's file system.
+ */
+export async function recoverFileSystem({
+  auth,
+  dependencies,
+  oldUsername,
+  newUsername,
+  readKey,
+}: {
+  auth: AuthenticationStrategy
+  dependencies: Dependencies
```

Comment:
I'd only ask for the components this function actually needs, instead of using the ones needed to load a file system. Less friction if people want to experiment.
```ts
import * as Crypto from "../components/crypto/implementation.js"
import * as Reference from "../components/reference/implementation.js"
import * as Storage from "../components/storage/implementation.js"

dependencies: {
  crypto: Crypto.Implementation
  reference: Reference.Implementation
  storage: Storage.Implementation
}
```

Reply:
ahh good point, will update!

Reply:
done 👍🏼